src/Dapr.Workflow/WorkflowState.cs

```diff
@@ -0,0 +1,132 @@
+
+        /// <summary>
+        /// Gets a value indicating whether the requested workflow is in a terminal state.
+        /// </summary>
+        public bool IsWorkflowCompleted => this.workflowState?.IsCompleted == true;
+
+        /// <summary>
+        /// Gets the time at which this workflow instance was created.
+        /// </summary>
+        public DateTimeOffset CreatedAt => this.workflowState?.CreatedAt ?? default;
+
+        /// <summary>
+        /// Gets the time at which this workflow instance last had its state updated.
+        /// </summary>
+        public DateTimeOffset LastUpdatedAt => this.workflowState?.LastUpdatedAt ?? default;
+
+        /// <summary>
+        /// Gets the execution status of the workflow.
+        /// </summary>
+        public WorkflowRuntimeStatus RuntimeStatus
+        {
+            get
+            {
+                if (this.workflowState == null)
+                {
+                    return WorkflowRuntimeStatus.Unknown;
+                }
+
+                switch (this.workflowState.RuntimeStatus)
+                {
+                    case OrchestrationRuntimeStatus.Running:
+                        return WorkflowRuntimeStatus.Running;
+                    case OrchestrationRuntimeStatus.Completed:
+                        return WorkflowRuntimeStatus.Completed;
+                    case OrchestrationRuntimeStatus.Failed:
+                        return WorkflowRuntimeStatus.Failed;
+                    case OrchestrationRuntimeStatus.Terminated:
+                        return WorkflowRuntimeStatus.Terminated;
+                    case OrchestrationRuntimeStatus.Pending:
+                        return WorkflowRuntimeStatus.Pending;
+                    default:
+                        return WorkflowRuntimeStatus.Unknown;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Deserializes the workflow input into <typeparamref name="T"/>.
+        /// </summary>
+        /// <typeparam name="T">The type to deserialize the workflow input into.</typeparam>
+        /// <returns>Returns the input as <typeparamref name="T"/>, or returns a default value if the workflow doesn't exist.</returns>
+        public T? ReadInputAs<T>()
+        {
+            if (this.workflowState == null)
+            {
+                return default;
+            }
+
+            return this.workflowState.ReadInputAs<T>();
+        }
+
+        /// <summary>
+        /// Deserializes the workflow output into <typeparamref name="T"/>.
+        /// </summary>
+        /// <typeparam name="T">The type to deserialize the workflow output into.</typeparam>
+        /// <returns>Returns the output as <typeparamref name="T"/>, or returns a default value if the workflow doesn't exist.</returns>
+        public T? ReadOutputAs<T>()
+        {
+            if (this.Exists)
+            {
+                return default;
+            }
```

Comment:
Is this `if` right? It looks reversed to me

Reply:
Whoops - good catch. I'll add output reading to the example so that this API gets some coverage.